tests/test_formatter.py

```diff
@@ -45,15 +45,15 @@ def testRegistry(self):
         formatterTypeName = "lsst.daf.butler.formatters.fitsCatalogFormatter.FitsCatalogFormatter"
         storageClassName = "Image"
         self.factory.registerFormatter(storageClassName, formatterTypeName)
-        f = self.factory.getFormatter(storageClassName)
+        f = self.factory.getFormatter(storageClassName, None)
```

Comment:
Are all of these `None`s really necessary here? Could you make wherever that is landing take a `None` default arg?

Reply:
I decided to force FileDescriptor to take an argument.  This at least made it obvious where I needed to put an argument during testing. It's more explicit this way that you are saying you know that the formatter will never be used to read or write anything.